tests/Doctrine/Tests/DBAL/Schema/MySqlSchemaManagerTest.php

```diff
@@ -37,6 +38,79 @@ public function testCompositeForeignKeys()
         $this->assertEquals(array('column_1', 'column_2', 'column_3'), array_map('strtolower', $fkeys[0]->getForeignColumns()));
     }
 
+    public function testPortableTableColumnDefinition()
+    {
+        $eventManager = new EventManager();
+        $driverMock = $this->createMock('Doctrine\DBAL\Driver');
+
+        $platforms = [
+            'mysql' => new \Doctrine\DBAL\Platforms\MySqlPlatform(),
+            'mariadb10.2' => new \Doctrine\DBAL\Platforms\MariaDb102Platform()
+        ];
+
+        foreach ($platforms as $platformName => $platform) {
+
+            $conn = $this->getMockBuilder('Doctrine\DBAL\Connection')
+                ->setMethods(array('fetchAll'))
+                ->setConstructorArgs(array(array('platform' => $platform), $driverMock, new Configuration(), $eventManager))
+                ->getMock();
+            $this->manager = new MySqlSchemaManager($conn);
+
+            $conn->expects($this->once())->method('fetchAll')->will($this->returnValue($this->getColumnDefinition($platform)));
+            $columns = $this->manager->listTableColumns('dummy');
+
+            $this->assertFalse($columns['id']->getNotnull());
+            $this->assertNull($columns['id']->getDefault());
+
+            $this->assertTrue($columns['created_by']->getNotnull());
+            $this->assertNull($columns['created_by']->getDefault());
+
+            $this->assertFalse($columns['text_1']->getNotnull());
+            $this->assertEquals('NULL', $columns['text_1']->getDefault());
+        }
+    }
+
+    public function getColumnDefinition($platform)
+    {
+        $mariaDb102 = ($platform instanceof MariaDb102Platform);
```

Comment:
Usually, it should be replaced with two different tests, one per platform. However, in this case, I don't think we should mock the DB at all.